connection.c

```diff
@@ -160,10 +204,7 @@ static void handleConnected(const redisAsyncContext *c, int status)
         return;
     }
 
-    /* Call explicit connect callback (even if failed) */
-    if (conn->connect_callback) {
-        conn->connect_callback(conn);
-    }
+    doAuthentication(conn, status);
```

Comment:
Shouldn't this be conditional, only if authentication is configured?

Reply:
ok done